components/server/ee/src/prebuilds/gitlab-app.ts

```diff
@@ -75,11 +75,11 @@ export class GitLabApp {
                 TraceContext.setError({ span }, error);
             }
             if (!user) {
-                // If the webhook installer is no longer found in Gitpod's DB
-                // we should send a UNAUTHORIZED signal.
+                // Gitpod is not supposed to return 4xx codes on issues with project permissions.
```

Comment:
Given this behaviour (200 on Error) is gonna look pretty bizzare to future readers, could you explain WHY it's doing it, rather than WHAT it's doing?

What I mean is to elaborate on why we cannot return a 4xx, and what are the implications if we do.

Reply:
That's a good question. The complete answer and docs are already contained at the top of this handler.

From the webhook event sender's perspective this is is not a 4xx case. We are not interested in a permanent disablement of the webhook if project permissions are temporarily "broken", instead prebuilds won't be executed until the maintainer had a chance to fix the issue. 

Also noteworthy, the response text is completely optional and helps us to understand the delivery logs on GitLab, but it's ignored otherwise. So this is the place where we may encode the App level error. 

The sender expects a 2xx response whenever it was a successful and legit delivery. It's up to the receiver to deal with issues of the application. For more on this see the linked issue and https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#failing-webhooks

Reply:
> The complete answer and docs are already contained at the top of this handler.

@AlexTugarev Looks like it only mentions 500. Can you add a line about 4xx as well? :pray:

Reply:
Thanks both of you for the hint. Just added some more details on the special behavior of webhooks, so that reading code makes more sense to future readers.